mypy/checkexpr.py

```diff
@@ -150,18 +150,13 @@ def analyze_ref_expr(self, e: RefExpr, lvalue: bool = False) -> Type:
         return result
 
     def analyze_var_ref(self, var: Var, context: Context) -> Type:
-        if not var.type:
+        if var.type:
+            return var.type
+        else:
```

Comment:
It would be worth calling out this "change" explicitly in the commit message, since it does not obviously look like a mere refactoring.